### PR TITLE
Feature/implement photo controller

### PR DESCRIPTION
**Pull request by Mmishaaa**

### CODE REVIEW

File: Tinder.API/Controllers/PhotoController.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    [ApiController]
+    public class PhotoController
+    {
+        private readonly IPhotoService _photoService;
+        private readonly IMapper _mapper;
+        public PhotoController(IPhotoService photoService, IMapper mapper)
+        {
+            _photoService = photoService;
+            _mapper = mapper;
+        }
```
Comment (moltenessence) [inline Tinder.API/Controllers/PhotoController.cs:22]: formatting

File: Tinder.API/Controllers/PhotoController.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        [HttpGet("{id}")]
+        public async Task<PhotoDto> GetByIdAsync(Guid userId, Guid id, CancellationToken cancellationToken)
+        {
+            var model = await _photoService.GetByIdAsync(userId, id, cancellationToken);
+            return _mapper.Map<PhotoDto>(model);
+        }
+
+        [HttpGet]
+
```
Comment (moltenessence) [inline Tinder.API/Controllers/PhotoController.cs]: formatting

File: Tinder.API/Controllers/PhotoController.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        [HttpPost]
+        public async Task<PhotoDto> Create(Guid userId, CreatePhotoDto dto, CancellationToken cancellationToken)
+        {
+            var photoToCreate = _mapper.Map<Photo>(dto);
+            var model = await _photoService.CreateAsync(userId, photoToCreate, cancellationToken);
+            return _mapper.Map<PhotoDto>(model);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<PhotoDto> Delete(Guid userId, Guid id, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline Tinder.API/Controllers/PhotoController.cs:47]: No unified style for HttpMethods attributes. In this case specify userId in route explicitly as well

File: Tinder.BLL/Services/PhotoService.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            await _photoRepository.DeleteAsync(entity, cancellationToken);
+            return _mapper.Map<Photo>(entity);
+        }
+
+        private Task<List<PhotoEntity>> UpdateUserPhotosAsync(User user, CancellationToken cancellationToken)
+        {
+            foreach (var userPhoto in user.Photos)
+            {
+                if (userPhoto.IsAvatar)
+                    userPhoto.IsAvatar = false;
```
Comment (moltenessence) [inline Tinder.BLL/Services/PhotoService.cs]: I do not get this logic

File: Tinder.DAL/Repositories/PhotoRepository.cs [hunk 5]
```diff
@@ -9,5 +10,58 @@ public PhotoRepository(ApplicationDbContext context) : base(context)
...
+        public override Task<List<PhotoEntity>> GetAllAsync(CancellationToken cancellationToken)
+        {
+            return _dbSet.AsNoTracking().Include(p => p.User).ToListAsync(cancellationToken);
+        }
+
+        public async Task<List<PhotoEntity>> UpdateRangeAsync(List<PhotoEntity> photos, CancellationToken cancellationToken)
+        {
+            foreach (var entity in photos)
+            {
+                var entityFromDb = await _dbSet.FirstOrDefaultAsync(p => p.Id == entity.Id, cancellationToken);
```
Comment (moltenessence) [inline Tinder.DAL/Repositories/PhotoRepository.cs]: 1000 photos = 1000 db queries. It it okay?

File: Tinder.DAL/Repositories/PhotoRepository.cs [hunk 6]
```diff
@@ -9,5 +10,58 @@ public PhotoRepository(ApplicationDbContext context) : base(context)
...
+                if (entityFromDb != null)
+                {
+                    entityFromDb.PhotoURL = entity.PhotoURL;
```
Comment (moltenessence) [inline Tinder.DAL/Repositories/PhotoRepository.cs]: I don't get the logic why you can replace one photo URL with another one

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: Tinder.BLL/Services/PhotoService.cs (hunk 4 above)
Comment (Mmishaaa) [reply Tinder.BLL/Services/PhotoService.cs]: if user uplaod a new photo with flag isAvatar = true, the previous avatar must become false

File: Tinder.DAL/Repositories/PhotoRepository.cs (hunk 5 above)
Comment (Mmishaaa) [reply Tinder.DAL/Repositories/PhotoRepository.cs]: zalupa, soglasen

File: Tinder.BLL/Services/PhotoService.cs (hunk 4 above)
Comment (moltenessence) [reply Tinder.BLL/Services/PhotoService.cs]: but why do you use cycle for this purpose? isAvatar can be set only for one user's photo. isn't it more logical to update only one entity?

**Review by VadimZharikov — APPROVED** [review]

File: Tinder.BLL/Services/PhotoService.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        public async Task<Photo> DeleteAsync(Guid userId, Guid id, CancellationToken cancellationToken)
+        {
+            var entity = await _photoRepository.GetByIdAsync(id, userId, cancellationToken) ?? throw new NotFoundException("Photo is not found");
+            await _photoRepository.DeleteAsync(entity, cancellationToken);
+            return _mapper.Map<Photo>(entity);
+        }
+
+        public async Task<Photo> UpdateAsync(Guid userId, Guid id, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline Tinder.BLL/Services/PhotoService.cs]: Name of the method is misleading as it doesn't say anything to me about avatar-specific logic

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Pull request merged**